src/game/UICmdList.cpp

```diff
@@ -0,0 +1,283 @@
+    for (int i = idCur; idCur < idEnd; idCur++, i++) {
+        if (idCur == 9000) {
+            i += 400; // offset and put gm cmds at end of map
+        } else if (idCur == 9100) {
+            i -= 500;
+            idCur = 9200;
+        }
+        szCommand.clear();
+        ::_LoadStringFromResource(idCur, szCommand);
+        if (!szCommand.empty() && (i / 100) % 2 == 0) {
+            m_mapCmds[i] = szCommand;
+        }
+    }
+
+    UpdateCommandList(CMD_LIST_PRIVATE); //initialize a cmd list for viewing when opening cmd window
+
+    return true;
+}
+bool CUICmdListDlg::UpdateCommandList(uint8_t cmdCat) {
+    if (m_pList_Cmds == NULL) {
+        return false;
+    }
+
+    m_pList_Cmds->ResetContent();
+
+    int indexStart = cmdCat * 200 + 8000; //start index for correct loc in map
+    int indexEnd = indexStart + 100;      //where to stop iterating
+    int i = 0;
+
+    for (auto itr = m_mapCmds.begin(); itr != m_mapCmds.end(); ++itr) {
+        if (itr->first >= indexStart && itr->first < indexEnd) {
+            m_pList_Cmds->AddString(itr->second);
+
+            CN3UIString * pChild = m_pList_Cmds->GetChildStrFromList(itr->second);
+            std::string   cmdTip;
+            ::_LoadStringFromResource(itr->first + 100, cmdTip);
+            if (pChild != NULL) {
+                pChild->SetTooltipText(cmdTip.c_str());
+            }
+            //SavvyNik tooltip is being loaded in but the rectangle
+            //that it shows on is too small. Need to figure out where
+            //this is being set.
+        }
+    }
+
+    return true;
+}
+bool CUICmdListDlg::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
+    if (NULL == pSender) {
+        return false;
+    }
+
+    if (dwMsg == UIMSG_BUTTON_CLICK) {
+        if (pSender == m_pBtn_cancel) {
+            SetVisible(false);
+            return true;
+        }
+    } else if (dwMsg == UIMSG_LIST_SELCHANGE) {
+        if (pSender == m_pList_CmdCat) {
+            uint8_t iSel = m_pList_CmdCat->GetCurSel();
+            UpdateCommandList(iSel);
+            return true;
+        }
+    } else if (dwMsg == UIMSG_LIST_DBLCLK) {
+        if (pSender == m_pList_Cmds) {
+            uint8_t iSel = m_pList_Cmds->GetCurSel();
+            ExecuteCommand(iSel);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void CUICmdListDlg::Render() {
+    if (!m_bVisible) {
+        return;
+    }
+
+    CN3UIBase::Render();
+}
+
+//this_ui_add_start
+bool CUICmdListDlg::OnKeyPress(int iKey) {
+    return CN3UIBase::OnKeyPress(iKey);
+}
+
+void CUICmdListDlg::SetVisible(bool bVisible) {
+    CN3UIBase::SetVisible(bVisible);
+    if (bVisible) {
+        CGameProcedure::s_pUIMgr->SetVisibleFocusedUI(this);
+    } else {
+        CGameProcedure::s_pUIMgr->ReFocusUI(); //this_ui
+    }
+}
+
+bool CUICmdListDlg::ExecuteCommand(uint8_t cmdSel) {
+    std::string command;
+    m_pList_Cmds->GetString(cmdSel, command);
+
+    if (command == "PM") {
```

Comment:
There's a couple of issues with doing this like this.
For one, you have the command names defined in the TBL, but then you're explicitly checking (and case-sensitive at that) it against "PM" here.

For another, it's not the only command which officially shows this prompt.

It would be saner to translate this back to either the command type from the enum, or alternatively the resource ID. Officially they do the former.